ai_diffusion/settings.py

```diff
@@ -8,17 +9,17 @@
 
 from . import util
 
-
 class ServerMode(Enum):
     undefined = -1
     managed = 0
     external = 1
 
 
 class ServerBackend(Enum):
-    cpu = "Run on CPU"
-    cuda = "Use CUDA (NVIDIA GPU)"
-    directml = "Use DirectML (GPU)"
+    cpu = ("Run on CPU", True)
+    cuda = ("Use CUDA (NVIDIA GPU)", sys.platform != 'darwin')
+    mps = ("Use MPS (Metal Performance Shader)", sys.platform == 'darwin')
+    directml = ("Use DirectML (GPU)", sys.platform.startswith("win"))
```

Comment:
Please use `util.is_windows`, and add a corresponding `util.is_macos`, so the check is in a central place.

Regarding the filtering I would suggest to build the list here:
```
@staticmethod
def supported():
    return [...] # OS-filtered list
```
Then instead of doing the filtering in various places, always use this list. Then there will be no index problems.

Reply:
util.is_windows is not usable there as util itself is importing setting, causing a circular import.

Reply:
Ok done. Please review.